HashRegistrarSimplified.sol

```diff
@@ -92,23 +92,49 @@ contract Deed {
     function () payable {}
 }
 
+
+/**
+ * @title Previous Registrar
+ * @dev The old registrar interface, for an upgrade path.
+ */
+contract PreviousRegistrar {
+    mapping (bytes32 => entry) public entries;
+
+    struct entry {
+        uint status;
+        Deed deed;
+        uint registrationDate;
+        uint value;
+        uint highestBid;
+    }
+
+    // Only exists for testing purposes
+    function register (bytes32 _hash) {
+        entry e = entries[_hash];
+        e.status = 2;
+        e.registrationDate = now;
+    }
+}
+
 /**
  * @title Registrar
  * @dev The registrar handles the auction process for each subnode of the node it owns.
  */
 contract Registrar {
     AbstractENS public ens;
     bytes32 public rootNode;
+    PreviousRegistrar public previousRegistrar;
```

Comment:
I think we need to keep this contract as the one initially deployed anywhere the auction contract is used, and put a version that does upgrades in a separate contract - better yet, have it extend the base registrar with upgrade functionality.

I don't think we can do what you're suggesting with using upgrades to swap out the 4 week auction for a 1 week auction, because that would require everyone who registered on the initial contract to manually migrate their deposits over.